resource_retriever/src/retriever.cpp

```diff
@@ -94,10 +94,10 @@ size_t curlWriteFunc(void * buffer, size_t size, size_t nmemb, void * userp)
   return size * nmemb;
 }
 
-MemoryResource Retriever::get(const std::string & url)
+std::string Retriever::resolve(const std::string & url)
 {
-  std::string mod_url = url;
   if (url.find("package://") == 0) {
```

Comment:
I know you didn't add this, but `url.find("package://")` seems too broad.  It will match lots of things that are probably not intended, like:

`/hey/this/is/a/path/that/happens/to/contain/package://foo/bar/baz`

Is it always the case that we expect the `package://` to be at the beginning of the string?  If so, I recently found a neat trick that emulates `str.startswith()`; it goes like:

```
url.rfind("package://", 0)
```

It works because in this signature, `rfind` starts at the position specified in the argument, and does the string compare.  If it doesn't find it, it walks backwards from that position until it hits 0.  (In C++ 20, I believe `startswith` was added, but that's too new for us right now).  What do you think about switching to something like that?

Reply:
I like the idea of requiring the beginning of the string to contain `package://`, and the `rfind` method seems reasonable.